common/nns.go

```diff
@@ -0,0 +1,47 @@
+package common
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop"
+	"github.com/nspcc-dev/neo-go/pkg/interop/contract"
+	"github.com/nspcc-dev/neo-go/pkg/interop/lib/address"
+	"github.com/nspcc-dev/neo-go/pkg/interop/native/management"
+	"github.com/nspcc-dev/neo-go/pkg/interop/native/std"
+)
+
+// NNSID is the ID of the NNS contract in NeoFS networks. It's always deployed
+// first.
+const NNSID = 1
+
+// ContractTLD is the default domain used by NeoFS contracts.
+const ContractTLD = "neofs"
+
+// InferNNSHash returns NNS contract hash by its well-known ID or panics if
+// it can't be resolved.
+func InferNNSHash() interop.Hash160 {
+	var nns = management.GetContractByID(1)
+	if nns == nil {
+		panic("no NNS contract")
+	}
+	return nns.Hash
+}
+
+// ResolveFSContract returns contract hash by name as registered in NNS or
+// panics if it can't be resolved.
+func ResolveFSContract(name string) interop.Hash160 {
```

Comment:
It is tested in deployment scenarios.

Reply:
fault conditions too? E.g. `hello world` instead of hash

Reply:
It's not a part of any public contract API, so a unit-test would require creating a specific contract to test just this thing. Push garbage in the deployment tests... Can be done, but I doubt it's useful. "hello world" is to be tried as an address and it will fail decoding miserably.